packages/server/lib/clients/oauth1.client.ts

```diff
@@ -137,6 +124,10 @@ export class OAuth1Client {
 
         const url = new URL(this.authConfig.authorization_url);
         const params = new URLSearchParams(queryParams);
-        return `${url}?${params.toString()}`;
+        return `${url.href}?${params.toString()}`;
     }
 }
+
+export function extractQueryParams(data: string | Buffer | undefined) {
+    return new URLSearchParams(typeof data === 'string' ? data : data?.toString());
+}
```

Comment:
why not add `Object.fromEntries` as part of the logic here?

Reply:
good point